### PR TITLE
deps(components): update @scion/components to Angular 16

### PR DESCRIPTION
**Pull request by Marcarrian**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [guidelines](https://github.com/SchweizerischeBundesbahnen/scion-toolkit/blob/master/CONTRIBUTING.md)
- [ ] Tests for the changes have been added
- [ ] Docs have been added or updated

## PR Type
What kind of change does this PR introduce?

- [ ] Fix
- [ ] Feature
- [ ] Documentation
- [ ] Refactoring (changes that neither fixes a bug nor adds a feature)
- [ ] Performance (changes that improve performance)
- [ ] Test (adding missing tests, refactoring tests; no production code change)
- [ ] Chore (other changes like formatting, updating the license, removal of deprecations, etc)
- [x] Deps (changes related to updating dependencies)
- [ ] CI (changes to our CI configuration files and scripts)
- [ ] Revert (revert of a previous commit)
- [ ] Release (publish a new release)
- [ ] Other... Please describe:

## Issue

Issue Number: #151 

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

### CODE REVIEW

File: projects/scion/components.e2e/src/sci-viewport/focus/viewport-focus.e2e-spec.ts [hunk 1]
```diff
@@ -8,7 +8,8 @@
  * SPDX-License-Identifier: EPL-2.0
  */
 
-import {expect, test} from '@playwright/test';
+import {test} from '../../fixtures';
```
Comment (danielwiehl) [inline projects/scion/components.e2e/src/sci-viewport/focus/viewport-focus.e2e-spec.ts:11]: Why did you change the import?

File: projects/scion/components.e2e/tsconfig.json [hunk 2]
```diff
@@ -1,11 +1,10 @@
...
-  "extends": "../../../tsconfig.json",
-  "compilerOptions": {
-    "outDir": "../../../out-tsc/components.e2e",
-    "module": "commonjs",
-    "target": "es2018",
-    "types": [
-      "node"
-    ]
+  "extends" : "../../../tsconfig.json",
+  "compilerOptions" : {
```
Comment (danielwiehl) [inline projects/scion/components.e2e/tsconfig.json:4]: Does this remove our import problems concerning ESModules in Playwright?

File: projects/scion/components.internal/package.json [hunk 3]
```diff
@@ -26,10 +26,10 @@
...
-    "@angular/common": "^15.0.0",
-    "@angular/core": "^15.0.0",
-    "@angular/cdk": "^15.0.0",
-    "@scion/components": "^15.0.0",
+    "@angular/common": "^16.0.0",
+    "@angular/core": "^16.0.0",
+    "@angular/cdk": "^16.0.0",
+    "@scion/components": "^16.0.0",
     "@scion/toolkit": "^1.0.0",
     "rxjs": "^7.5.0"
```
Comment (danielwiehl) [inline projects/scion/components.internal/package.json:34]: I would change it to `^7.8.0` because this is the minimal version Angular requires.

File: projects/scion/components.internal/package.json [hunk 4]
```diff
@@ -38,5 +38,6 @@
   "repository": {
     "type": "git",
     "url": "https://github.com/SchweizerischeBundesbahnen/scion-toolkit.git"
-  }
+  },
+  "sideEffects": false
```
Comment (danielwiehl) [inline projects/scion/components.internal/package.json:42]: Do we have to configure this property? What is the default? I cannot imagine that the default is `true`.

File: projects/scion/components.internal/list/src/list.component.scss [hunk 5]
```diff
@@ -20,6 +20,10 @@
     border: 1px solid var(--sci-color-P400);
     border-radius: 5px;
 
+    &::part(content) {
+      grid-auto-rows: min-content;
```
Comment (danielwiehl) [inline projects/scion/components.internal/list/src/list.component.scss:24]: Why not changing the layout to flex-column?

**Review by danielwiehl — CHANGES_REQUESTED** [review]

File: projects/scion/components.internal/package.json (hunk 4 above)
Comment (Marcarrian) [reply projects/scion/components.internal/package.json:42]: Set when generating a new Angular library

**Review by danielwiehl — APPROVED** [review]
